src/plugins/intel_gpu/src/plugin/transformations/swiglu_fusion.cpp

```diff
@@ -0,0 +1,87 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "swiglu_fusion.hpp"
+
+#include "intel_gpu/op/swiglu.hpp"
+
+#include "openvino/core/rt_info.hpp"
+#include "openvino/op/constant.hpp"
+#include "openvino/op/multiply.hpp"
+#include "openvino/op/swish.hpp"
+#include "openvino/op/variadic_split.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "transformations/utils/utils.hpp"
+
+namespace ov {
+namespace intel_gpu {
+
+SwiGLUFusion::SwiGLUFusion() {
+    using namespace ov::pass::pattern;
+
+    // Detect SwiGLU decomposition pattern
+    // SwiGLU(Xw, Xv, beta) = (Xw * (1.0 + exp(-beta * Xw))) * Xv
+    auto data = any_input();
+
+    // VariadicSplit(X, axis, split_lengths) = Xw, Xv
+    auto const_axis = wrap_type<ov::op::v0::Constant>();
+    auto const_split_lengths = wrap_type<ov::op::v0::Constant>();
+    auto variadic_split = wrap_type<ov::op::v1::VariadicSplit>({data, const_axis, const_split_lengths});
+
+    // Swish(Xw) = Xw * (1.0 + exp(-beta * Xw))
+    auto swish = wrap_type<ov::op::v4::Swish>({variadic_split});
+
+    // Mul(Xw, Xv) = Swish(Xw) * Xv
+    auto mul = wrap_type<ov::op::v1::Multiply>({swish, variadic_split});
```

Comment:
I think the pattern should use different output ports of variadic split:
```cpp
    // Swish(Xw) = Xw * (1.0 + exp(-beta * Xw))
    auto swish = wrap_type<ov::op::v4::Swish>({variadic_split->output(0)});

    // Mul(Xw, Xv) = Swish(Xw) * Xv
    auto mul = wrap_type<ov::op::v1::Multiply>({swish, variadic_split->output(1)});
```

Otherwise the pattern where both nodes use same port will also be replaced by SwiGLU op which is not expected. Please also add a test case for that (in pass unit test)

Reply:
Updated pattern that complements your concern and added test case to check corresponding scenario